src/main/kotlin/com/team4099/robot2023/commands/drivetrain/TargetPoseCommand.kt

```diff
@@ -0,0 +1,156 @@
+    val driver: DriverProfile,
+    val driveX: () -> Double,
+    val driveY: () -> Double,
+    val turn: () -> Double,
+    val slowMode: () -> Boolean,
+    val drivetrain: Drivetrain,
+    targetCoordinate: FrameCoordinate
+) : Command() {
+    private val thetaPID: ProfiledPIDController<Radian, Velocity<Radian>>
+    private val fieldVelocitySupplier = { drivetrain.fieldVelocity.x to drivetrain.fieldVelocity.y }
+    val targetCoordinate = if (FMSData.allianceColor == DriverStation.Alliance.Red && targetCoordinate is FrameCoordinate.FieldCoordinate) {
+        FrameCoordinate.FieldCoordinate(FieldConstants.fieldLength - targetCoordinate.x, targetCoordinate.y)
+    } else targetCoordinate
+
+    val thetakP =
+        LoggedTunableValue(
+            "Pathfollow/thetakP",
+            Pair({ it.inDegreesPerSecondPerDegree }, { it.degrees.perSecond.perDegree })
+        )
+    val thetakI =
+        LoggedTunableValue(
+            "Pathfollow/thetakI",
+            Pair(
+                { it.inDegreesPerSecondPerDegreeSeconds }, { it.degrees.perSecond.perDegreeSeconds }
+            )
+        )
+    val thetakD =
+        LoggedTunableValue(
+            "Pathfollow/thetakD",
+            Pair(
+                { it.inDegreesPerSecondPerDegreePerSecond },
+                { it.degrees.perSecond.perDegreePerSecond }
+            )
+        )
+
+    val thetaMaxVel =
+        LoggedTunableValue("Pathfollow/thetaMaxVel", DrivetrainConstants.PID.MAX_AUTO_ANGULAR_VEL)
+    val thetaMaxAccel =
+        LoggedTunableValue("Pathfollow/thetaMaxAccel", DrivetrainConstants.PID.MAX_AUTO_ANGULAR_ACCEL)
+    var desiredAngle: Angle = 0.0.degrees
+    lateinit var odometryCoordinateToTarget: FrameCoordinate.OdometryCoordinate
+
+    init {
+        addRequirements(drivetrain)
+
+        if (RobotBase.isReal()) {
+            thetakP.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KP)
+            thetakI.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KI)
+            thetakD.initDefault(DrivetrainConstants.PID.AUTO_THETA_PID_KD)
+        } else {
+            thetakP.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KP)
+            thetakI.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KI)
+            thetakD.initDefault(DrivetrainConstants.PID.SIM_AUTO_THETA_PID_KD)
+        }
+
+        thetaPID =
+            ProfiledPIDController(
+                thetakP.get(),
+                thetakI.get(),
+                thetakD.get(),
+                TrapezoidProfile.Constraints(thetaMaxVel.get(), thetaMaxAccel.get())
+            )
+        thetaPID.enableContinuousInput(-PI.radians, PI.radians)
+    }
+
+    override fun initialize() {
+        thetaPID.reset(drivetrain.odomTRobot.rotation)
+
+        odometryCoordinateToTarget = if (targetCoordinate is FrameCoordinate.FieldCoordinate) {
+            FrameCoordinate.OdometryCoordinate(
+                drivetrain.odomTField.asPose2d().transformBy(targetCoordinate.toTransform2d())
```

Comment:
i would prefer we do this in the FrameCoordinate class

Reply:
you should do the allianceUtil flip stuff here though before you convert the FieldCoordinate to an odometry one